lib/datadog/appsec/contrib/graphql/gateway/multiplex.rb

```diff
@@ -28,41 +28,74 @@ def queries
 
             private
 
-            def create_arguments_hash
-              args = {}
-              @multiplex.queries.each_with_index do |query, index|
-                resolver_args = {}
-                resolver_dirs = {}
-                selections = (query.selected_operation.selections.dup if query.selected_operation) || []
-                # Iterative tree traversal
-                while selections.any?
-                  selection = selections.shift
-                  set_hash_with_variables(resolver_args, selection.arguments, query.provided_variables)
-                  selection.directives.each do |dir|
-                    resolver_dirs[dir.name] ||= {}
-                    set_hash_with_variables(resolver_dirs[dir.name], dir.arguments, query.provided_variables)
-                  end
-                  selections.concat(selection.selections)
+            # This method builds an array of argument hashes for each field with arguments in the query.
+            #
+            # For example, given the following query:
+            # query ($postSlug: ID = "my-first-post", $withComments: Boolean!) {
+            #   firstPost: post(slug: $postSlug) {
+            #     title
+            #     comments @include(if: $withComments) {
+            #       author { name }
+            #       content
+            #     }
+            #   }
+            # }
+            #
+            # The result would be:
+            # {"post"=>[{"slug"=>"my-first-post"}], "comments"=>[{"include"=>{"if"=>true}}]}
+            #
+            # Note that the `comments` "include" directive is included in the arguments list
+            def build_arguments_hash
+              queries.each_with_object({}) do |query, args_hash|
+                next unless query.selected_operation
+
+                arguments_from_selections(query.selected_operation.selections, query.variables, args_hash)
+              end
+            end
+
+            def arguments_from_selections(selections, query_variables, args_hash)
+              selections.each do |selection|
+                next unless selection.is_a?(::GraphQL::Language::Nodes::Field)
+
+                selection_name = selection.alias || selection.name
+
+                if selection.arguments.any? || selection.directives.any?
```

Comment:
Do you use `any?` here because we actually need to check if there are some non-nil elements in the array or would checking if array is empty enough?

`any?` implicitly [constructs a block and iterates the enumerable](https://apidock.com/ruby/Enumerable/any%3F), `empty?` only checks the array length (so it is almost a single instruction operation)

Reply:
it should be enough to check if it is empty actually, thanks!